### PR TITLE
INT: add specify type explicitly intention

### PR DESCRIPTION
**Pull request by sirgl**
peek 2017-06-11 18-31

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/intentions/SpecifyTypeExplicitlyIntention.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    override fun getText() = "Specify type explicitly"
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        val letDecl = element.parentOfType<RsLetDecl>() ?: return null
+        if(letDecl.typeReference != null) {
+            return null
+        }
+        val ident = letDecl.pat as? RsPatIdent ?: return null
+        val type: Ty = inferDeclarationType(ident.patBinding)
```
Comment (mkaput) [inline src/main/kotlin/org/rust/ide/intentions/SpecifyTypeExplicitlyIntention.kt]: heh, the explicit type declaration here is unnecessary 😉

File: src/main/kotlin/org/rust/ide/intentions/SpecifyTypeExplicitlyIntention.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        return Context(type, letDecl)
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: Context) {
+        val factory = RsPsiFactory(project)
+        val createdType = factory.createType(ctx.type.toString())
+        val letDecl = ctx.letDecl
+        val colon = letDecl.addAfter(factory.createColon(), letDecl.pat)
+        colon.parent.addAfter(createdType, colon)
```
Comment (mkaput) [inline src/main/kotlin/org/rust/ide/intentions/SpecifyTypeExplicitlyIntention.kt]: why not just `letDecl` instead of `colon.parent`?

**Comment by sirgl** [comment]
I think shorter means better :)

**Comment by mkaput** [comment]
I think this is small enough that I can shamelessly type it for the first time :D

bors r+

Thanks!

**Pull request merged**